python/lsst/pipe/analysis/utils.py

```diff
@@ -41,11 +42,53 @@
            "writeParquet", "getRepoInfo", "findCcdKey", "getCcdNameRefList", "getDataExistsRefList",
            "orthogonalRegression", "distanceSquaredToPoly", "p1CoeffsFromP2x0y0", "p2p1CoeffsFromLinearFit",
            "lineFromP2Coeffs", "linesFromP2P1Coeffs", "makeEqnStr", "catColors", "setAliasMaps",
-           "addPreComputedColumns", "addMetricMeasurement", "updateVerifyJob", "computeMeanOfFrac"]
+           "addPreComputedColumns", "addMetricMeasurement", "updateVerifyJob", "computeMeanOfFrac",
+           "persistPlots"]
 
 
 NANOJANSKYS_PER_AB_FLUX = (0*units.ABmag).to_value(units.nJy)
 
+
+def persistPlots(plotList, dataId, butler):
+    """Persist plots and parse stats yielded by the supplied generator
+
+    Parameters
+    ----------
+    plotList : `list`
+        List of generators that will yield plots
+    dataId : `dict`
+        The dataId that will be used in persisting plots
+    butler : `lsst.daf.persistence.Butler`
+        The butler that might be used in persisting the plots
+    """
+    allStats = {}
+    allStatsHigh = {}
+    n = 0
+    for plots in plotList:
+        # In this conext next is called to "prime" the generator
+        # Each plotting function yields right after it is called so that
+        # each python execution frame is created, but not run. Calling next
+        # on the reference to the plot function begins the execution of the
+        # body, which will then yield each plot the function creates.
+        next(plots)
+        for plot in plots:
+            if hasattr(plot, 'description'):
+                dataId['description'] = plot.description
+                key = plot.description
+                butler.put(plot.fig, "plotFakes", dataId)
+            elif hasattr(plot, 'filename'):
+                key = plot.filename
+                plot.fig.savefig(plot.filename, dpi=120)
```

Comment:
Can you save @laurenam's special dpi's by adding it as an optional element of the Struct?